src/plugins/dashboard/public/application/actions/export_csv_action.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Datatable } from 'src/plugins/expressions/public';
+import { FormatFactory } from '../../../../data/common/field_formats/utils';
+import { DataPublicPluginStart, exporters } from '../../../../data/public';
+import { downloadMultipleAs } from '../../../../share/public';
+import { Adapters, IEmbeddable } from '../../../../embeddable/public';
+import { ActionByType } from '../../../../ui_actions/public';
+import { CoreStart } from '../../../../../core/public';
+
+export const ACTION_EXPORT_CSV = 'ACTION_EXPORT_CSV';
+
+export interface Params {
+  core: CoreStart;
+  data: DataPublicPluginStart;
+}
+
+export interface ExportContext {
+  embeddable?: IEmbeddable;
+  // used for testing
+  asString?: boolean;
+}
+
+/**
+ * This is "Export CSV" action which appears in the context
+ * menu of a dashboard panel.
+ */
+export class ExportCSVAction implements ActionByType<typeof ACTION_EXPORT_CSV> {
+  public readonly id = ACTION_EXPORT_CSV;
+
+  public readonly type = ACTION_EXPORT_CSV;
+
+  public readonly order = 200;
```

Comment:
cc @ThomThomson on this but this high priority order pushed "Edit panel title" into the "More" part of the context menu (Visualize vs Lens):
<img width="299" alt="Screenshot 2020-11-30 at 14 47 44" src="https://user-images.githubusercontent.com/1508364/100617671-1f77a380-331b-11eb-9668-622d1c1df53c.png">

<img width="295" alt="Screenshot 2020-11-30 at 14 47 40" src="https://user-images.githubusercontent.com/1508364/100617668-1dade000-331b-11eb-828c-c51e0bf2c2c2.png">

To keep consistency and because I expect changing the title will be a much more common operation than downloading the csv, let's not show it that high in the list.

Reply:
Moved the action to `order = 20`